backend/src/main/java/com/redhat/cloud/notifications/db/repositories/TemplateRepository.java

```diff
@@ -0,0 +1,338 @@
+    }
+
+    public List<AggregationEmailTemplate> findAggregationEmailTemplatesByApplication(UUID appId) {
+        String hql = "FROM AggregationEmailTemplate t JOIN FETCH t.application a WHERE a.id = :appId";
+        List<AggregationEmailTemplate> aggregationEmailTemplates = entityManager.createQuery(hql, AggregationEmailTemplate.class)
+                .setParameter("appId", appId)
+                .getResultList();
+        for (AggregationEmailTemplate aggregationEmailTemplate : aggregationEmailTemplates) {
+            // The full application isn't needed in the REST response.
+            aggregationEmailTemplate.filterOutApplication();
+            // The full templates aren't needed in the REST response.
+            aggregationEmailTemplate.filterOutTemplates();
+        }
+        return aggregationEmailTemplates;
+    }
+
+    public AggregationEmailTemplate findAggregationEmailTemplateById(UUID id) {
+        String hql = "FROM AggregationEmailTemplate t JOIN FETCH t.subjectTemplate JOIN FETCH t.bodyTemplate " +
+                "LEFT JOIN FETCH t.application WHERE t.id = :id";
+        try {
+            AggregationEmailTemplate template = entityManager.createQuery(hql, AggregationEmailTemplate.class)
+                    .setParameter("id", id)
+                    .getSingleResult();
+            // The full application isn't needed in the REST response.
+            template.filterOutApplication();
+            return template;
+        } catch (NoResultException e) {
+            throw new NotFoundException("Aggregation email template not found");
+        }
+    }
+
+    @Transactional
+    public boolean updateAggregationEmailTemplate(UUID id, AggregationEmailTemplate template) {
+        String hql = "UPDATE AggregationEmailTemplate SET application = :app, subjectTemplate = :subjectTemplate, " +
+                "subscriptionType = :subscriptionType, bodyTemplate = :bodyTemplate WHERE id = :id";
+
+        Application app;
+        if (template.getApplicationId() == null) {
+            app = null;
+        } else {
+            app = findApplication(template.getApplicationId());
+        }
+        Template subjectTemplate = findTemplate(template.getSubjectTemplateId(), SUBJECT_NOT_FOUND);
+        Template bodyTemplate = findTemplate(template.getBodyTemplateId(), BODY_NOT_FOUND);
+
+        int rowCount = entityManager.createQuery(hql)
+                .setParameter("app", app)
+                .setParameter("subscriptionType", template.getSubscriptionType())
+                .setParameter("subjectTemplate", subjectTemplate)
+                .setParameter("bodyTemplate", bodyTemplate)
+                .setParameter("id", id)
+                .executeUpdate();
+        return rowCount > 0;
+    }
+
+    @Transactional
+    public boolean enableOrDisableAggregationEmailTemplate(UUID id, boolean enabled) {
+        if (enabled) {
+            AggregationEmailTemplate emailTemplate = findAggregationEmailTemplate(id);
```

Comment:
Isn't this throwing if nothing is enabled?

Reply:
The idea here, which I wanted to discuss with you, is to automatically disable other templates before enabling a new one because we can't have multiple templates enabled at the same time. There is a similar code for instant email template.

Another way of dealing with multiple enabled templates would be an exception throw that would be forwarded to the frontend and cancel the user action, asking him to first disable the other template before enabling a new one.

I have a slight preference for the current code which would only require a warning in the frontend when the user is about to enable a template, saying something like:
```
Warning. If you enable this template, all other templates that are already associated with [app] / [subscription type] will be automatically disabled. Do you want to continue? Yes/No
```

Or, the frontend could first call the backend and run a `COUNT(*)` on templates to check if there is another enabled template linked to the app/subscriptionType and only display the warning if there is one.

Reply:
Or maybe we don't need the `enabled` field at all and linking the template to the app/subscriptionType is enough to enable a template 😄

Plenty of options here, we can pick any of them and I'll adapt the code.

Reply:
I think it's better to link-to-enable, and only 1 can be linked at a time.

Reply:
I pushed a new commit and removed the `enabled` field from the email templates.